editoast/editoast_schemas/src/rolling_stock.rs

```diff
@@ -87,3 +93,78 @@ pub struct RollingStock {
     #[serde(default)]
     pub metadata: Option<RollingStockMetadata>,
 }
+
+impl<'de> Deserialize<'de> for RollingStock {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct Inner {
+            pub name: String,
+            pub locked: bool,
+            pub effort_curves: EffortCurves,
+            pub base_power_class: Option<String>,
+            pub length: f64,
+            pub max_speed: f64,
+            pub startup_time: f64,
+            pub startup_acceleration: f64,
+            pub comfort_acceleration: f64,
+            pub const_gamma: f64,
+            pub etcs_brake_params: Option<EtcsBrakeParams>,
+            pub inertia_coefficient: f64,
+            pub mass: f64,
+            pub rolling_resistance: RollingResistance,
+            pub loading_gauge: LoadingGaugeType,
+            #[serde(default)]
+            pub power_restrictions: HashMap<String, String>,
+            #[serde(default)]
+            pub energy_sources: Vec<EnergySource>,
+            pub electrical_power_startup_time: Option<f64>,
+            #[serde(default)]
+            pub raise_pantograph_time: Option<f64>,
+            pub supported_signaling_systems: RollingStockSupportedSignalingSystems,
+            pub railjson_version: String,
+            #[serde(default)]
+            pub metadata: Option<RollingStockMetadata>,
+        }
+
+        let inner = Inner::deserialize(deserializer)?;
+
+        if inner
+            .supported_signaling_systems
+            .0
+            .contains(&"ETCS_LEVEL2".to_string())
+            && inner.etcs_brake_params.is_none()
```

Comment:
This validation makes sense, but I realize we already have some validation of coherence of the input data, in [a different place](https://github.com/OpenRailAssociation/osrd/blob/7582bddb071ae51efa1c5b13039ea3c59dd1307b/editoast/src/models/rolling_stock_model.rs#L93-L140). At the very least, I would expect all of these validations to happen at the same place.

I can understand why we did it somewhere else: in a 3-tier architecture, deserialization should only check if it's a valid considering the protocol (JSON in this case), not if the data inside is coherent relatively to the business logic. Then the model which will be inserted into DB has to be checked logically too, so conversion from `RollingStock` to `RollingStockModel` will do this verification.

That said, I'm also a fervent defender [Parse, don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/), basically, validate early and don't think about it anymore (and hopefully, don't forget to call the `.validate_imported_rolling_stock()` function. cc @leovalais 

In any case, I believe we should move everything to one place or the other in the end.